packages/jwt-verifier/lib.js

```diff
@@ -115,15 +139,19 @@ class OktaJwtVerifier {
     });
     this.verifier = nJwt.createVerifier().setSigningAlgorithm('RS256').withKeyResolver((kid, cb) => {
       this.jwksClient.getSigningKey(kid, (err, key) => {
+        // If kid is undefined, but there is 1 signing key, `jwks-rsa` will not throw error
+        if (!kid)
```

Comment:
is there a JIRA issue related to this undefined kid condition? This seems like its fixing a bug. Should it be in a separate commit?
Also, would it be better to check this condition before calling `getSigningKey` ?

Reply:
There is [a test for this condition](https://github.com/okta/okta-oidc-js/blob/master/packages/jwt-verifier/test/spec/verifier.spec.js#L100).
But currently, it depends on the configuration of the authorization server. 
If there are 2 public keys in JWKS, `jwks-rsa` will [throw the error](https://github.com/auth0/node-jwks-rsa/blob/4446484a5373d7a65873616a7c4d7889e48f00f3/src/JwksClient.js#L103), which will result in [error from njwt](https://github.com/jwtk/njwt/blob/d6f81191addfdbee5d146bd4a6f3e8b783802b41/index.js#L379):  "Error while resolving signing key for kid  undefined"
If there is 1 key in JWKS, no error will be thrown and tests will fail (which happens currently)

Reply:
Maybe remove code change and skip test `should fail if no kid is present in the JWT header`?

Reply:
If "kid" is required, why don't we throw before calling `getSigningKey`? Is there another condition we are missing here? Should it be

```
if (!kid && !err)
```
or
```
if (!id && key) // this seems to match the comment? (kid is undefined, but there is 1 signing key)
```

The other question is if you think this is a bug in jwk-rsa or njwt? If so we can include the fix here but in a comment reference another issue to fix it in the other library.

Reply:
I latest commit I've changed behavior to [throw an error if KID is not specified](https://github.com/okta/okta-oidc-js/pull/951/files#diff-944b6965a1656adcd833a6937e8d2e7328110332310c7a607e225b0097b487d8R146) , before calling `getSigningKey `